packages/cactus-plugin-ledger-connector-xdai/src/main/typescript/plugin-ledger-connector-xdai.ts

```diff
@@ -478,34 +478,38 @@ export class PluginLedgerConnectorXdai
     consistencyStrategy: ConsistencyStrategy,
   ): Promise<TransactionReceipt> {
     const fnTag = `${this.className}#pollForTxReceipt()`;
-    let txReceipt;
-    let timedOut = false;
-    let tries = 0;
-    let confirmationCount = 0;
-    const timeoutMs = consistencyStrategy.timeoutMs || Number.MAX_SAFE_INTEGER;
-    const startedAt = new Date();
-
-    do {
-      tries++;
-      timedOut = Date.now() >= startedAt.getTime() + timeoutMs;
-      if (timedOut) {
-        break;
-      }
-
-      txReceipt = await this.web3.eth.getTransactionReceipt(txHash);
-      if (!txReceipt) {
-        continue;
-      }
-
-      const latestBlockNo = await this.web3.eth.getBlockNumber();
-      confirmationCount = latestBlockNo - txReceipt.blockNumber;
-    } while (confirmationCount >= consistencyStrategy.blockConfirmations);
-
-    if (!txReceipt) {
-      throw new Error(`${fnTag} Timed out ${timeoutMs}ms, polls=${tries}`);
-    }
-    return txReceipt;
-  }
+    return new Promise<TransactionReceipt>(async( resolve: (value: TransactionReceipt | PromiseLike<TransactionReceipt>) => void, reject: (reason?: any) => void)=>{
+      if (consistencyStrategy.receiptType === ReceiptType.NodeTxPoolAck){
+        consistencyStrategy.blockConfirmations = 0
+      }  
+      const intervalMs = consistencyStrategy.callInterval
+        const timeoutMs = consistencyStrategy.timeoutMs || Number.MAX_SAFE_INTEGER
+        const timeoutAt = (new Date()).getTime() + timeoutMs
+        let tries = 0
+        const interval = setInterval(async ()=>{
```

Comment:
@jagpreetsinghsasan I recommend using a one-liner to implement the wait instead of an explicit setInterval call wrapping everything.
It ends up being the same under the hood but this way the change becomes much smaller, easier to read.
The one liner I usually do to "sleep" goes like this
```typescript
await new Promise((resolve) => setInterval(resolve, pollIntervalMs));
```

Reply:
something like this ?

```typescript
  public async pollForTxReceipt(
    txHash: string,
    consistencyStrategy: ConsistencyStrategy,
  ): Promise<TransactionReceipt> {
    const fnTag = `${this.className}#pollForTxReceipt()`;
    const intervalMs = consistencyStrategy.callInterval
    let receipt:TransactionReceipt
    try {
      receipt = await new Promise<TransactionReceipt>((resolve: (value: TransactionReceipt ) => void, reject: (reason?: any) => void)=>{
          setInterval(async()=>{
            // fetch tx Receipt
            // check block confirmations
            // check timeout, if timed out call reject
          },intervalMs)
      })
    } catch (error) {
      throw error
    }
    return receipt
}
```

Reply:
@Zzocker No, you'd just put in the one-liner to sleep in the body of while loop